adagrams/game.py

```diff
@@ -1,11 +1,166 @@
+            continue
+        else:
+            return False
+        
+            
+    return True
+
 
 def score_word(word):
-    pass
+    LETTER_SCORE = {'A': 1,'B': 3,'C': 3,'D': 2,'E': 1,'F': 4,'G': 2,'H': 4,'I': 1,'J': 8,'K': 5, 
+    'L': 1,'M': 3,'N': 1,'O': 1,'P': 3,'Q': 10,'R': 1,'S': 1,'T': 1,'U': 1,'V': 4,'W': 4,'X': 8,'Y': 4,'Z':10}
+    word_upper = word.upper()
+    score = 0
+        
+    for letter in word_upper:
+        if letter in LETTER_SCORE:
+            score += LETTER_SCORE[letter]
+    if len(word_upper) >= 7 and len(word_upper) <= 10:
+            score += 8
+    return score
 
+#CODED_BY_YUN:
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    sorted_word_list = sorted(word_list, key=len)
+    score_list = []
+    word_score_dict = {}
+    for word in sorted_word_list:
+        word_score = score_word(word)
+        score_list.append(word_score)
+        word_score_dict[word] = word_score
+
+    highest_word_score = max(set(score_list))
+    highest_word_dict = {}
+    for word, score in word_score_dict.items():
+        if score == highest_word_score:
+            highest_word_dict[word] = score
+    for word, score in highest_word_dict.items():
+        if len(word) == 10:
+            winning_tuple = (word, score)
+            break
+        else:
+            winning_tuple = list(highest_word_dict.items())[0]
+    return winning_tuple
+
+##CODED_BY_IS##
+# def get_highest_word_score(word_list):
+#     word_to_score_dict = {}
+#     for word in word_list:
+#         word_score = score_word(word)
+#         word_to_score_dict[word] = word_score
+#     winning_words = []
+#     highest_scoring_word = ()
+#     highest_scoring_word = (word, word_score)
+    
+#     for word, word_score in word_to_score_dict.items():
+#         if highest_scoring_word == ():
+#             highest_scoring_word = (word, word_score)
+#             continue
+#         if word_score > highest_scoring_word[1]:
+#             highest_scoring_word = (word, word_score)
+#             winning_words.append(highest_scoring_word)
+        
+#         if word_score == highest_scoring_word[1]:
+#             if len(highest_scoring_word[0]) == 10:
+#                 winning_words.append(highest_scoring_word)
+#                 return highest_scoring_word
+
+#             if len(word) == 10 and len(highest_scoring_word[0]) <= len(word):
+#                 highest_scoring_word = (word, word_to_score_dict[word])
+#                 winning_words.append(highest_scoring_word)
+#                 return highest_scoring_word
+#             if len(highest_scoring_word[0]) > len(word):
+#                 highest_scoring_word = (word, word_to_score_dict[word])
+#             elif len(highest_scoring_word[0]) == len(word):
+#                 highest_scoring_word = (word, word_to_score_dict[word])
+            
+#             winning_words.append(highest_scoring_word)
+#             return highest_scoring_word
```

Comment:
There's a lot of great logic here! I think the main issue this code has is the return statements being inside the `for` loop. A return statement immediately ends our loop and exits our function, which means that none of our later words would be considered. Moving our return statement outside the loop would solve. lot here!